process/block/shardblock.go

```diff
@@ -279,36 +283,50 @@ func (sp *shardProcessor) ProcessBlock(
 	return nil
 }
 
+func (sp *shardProcessor) requestEpochStartInfo(header *block.Header, waitTime time.Duration) error {
+	sp.mutEpochStart.Lock()
+	_ = process.EmptyChannel(sp.chRcvEpochStart)
+	haveMissingMetaHeaders := header.IsStartOfEpochBlock() && sp.epochStartTrigger.IsEpochStart() == false
+	sp.mutEpochStart.Unlock()
+
+	if haveMissingMetaHeaders {
+		select {
+		case <-sp.chRcvEpochStart:
+			return nil
+		case <-time.After(waitTime):
+			return process.ErrTimeIsOut
+		}
+	}
+
+	return nil
+}
+
 func (sp *shardProcessor) checkEpochCorrectness(
-	headerHandler data.HeaderHandler,
+	header *block.Header,
 	chainHandler data.ChainHandler,
 ) error {
 	currentBlockHeader := chainHandler.GetCurrentBlockHeader()
 	if currentBlockHeader == nil {
 		return nil
 	}
 
-	isEpochIncorrect := headerHandler.GetEpoch() != currentBlockHeader.GetEpoch() &&
+	isEpochIncorrect := header.GetEpoch() != currentBlockHeader.GetEpoch() &&
 		sp.epochStartTrigger.Epoch() == currentBlockHeader.GetEpoch()
 	if isEpochIncorrect {
 		return process.ErrEpochDoesNotMatch
 	}
 
 	isOldEpochAndShouldBeNew := sp.epochStartTrigger.IsEpochStart() &&
-		headerHandler.GetRound() > sp.epochStartTrigger.EpochStartRound()+process.MetaBlockFinality &&
-		headerHandler.GetEpoch() != sp.epochStartTrigger.Epoch()
+		header.GetRound() > sp.epochStartTrigger.EpochStartRound()+process.MetaBlockFinality+1 &&
```

Comment:
what is the +1 for?

Reply:
metachain accepts blocks from shard even in epoch change if their round is smaller than epoch change round + finality + 1. The same is here. Maybe the leader did not get the metaheaders, so he doesn't no yet there was an epoch change.

Reply:
let's name this instead and take it either as from config if this is dependent on other config parameters and may change or have a constant.

Reply:
it is named. Added constant in process.